temba/api/tests.py

```diff
@@ -3166,8 +3183,15 @@ def test_send(self):
                 self.assertEquals('1234567890', msg.external_id)
                 self.assertTrue(msg.sent_on)
 
-                self.clear_cache()
+    def test_send_error(self):
+        # This tests a send error.
+        joe = self.create_contact("Joe", number="+250788383383", twitter="joe1981")
+        bcast = joe.send("Test message", self.admin, trigger_send=False)
+
+        # our outgoing message
+        msg = bcast.get_messages()[0]
```

Comment:
Fair enough. Admittedly these are some serious utility tests that were generally written in a rush as we were integrating a new partner, so ya, they could use some more love to normalize it all.

But you are definitely leaving them better than you found them. :+1: